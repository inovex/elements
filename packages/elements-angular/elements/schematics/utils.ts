import { virtualFs, workspaces } from '@angular-devkit/core';
import { SchematicsException, Tree } from '@angular-devkit/schematics';

const CONFIG_PATH = 'angular.json';

function createHost(tree: Tree): workspaces.WorkspaceHost {
  return {
    async readFile(path: string): Promise<string> {
      const data = tree.read(path);
      if (!data) {
        throw new SchematicsException('File not found.');
      }
      return virtualFs.fileBufferToString(data);
    },
    async writeFile(path: string, data: string): Promise<void> {
      return tree.overwrite(path, data);
    },
    async isDirectory(path: string): Promise<boolean> {
      return !tree.exists(path) && tree.getDir(path).subfiles.length > 0;
    },
    async isFile(path: string): Promise<boolean> {
      return tree.exists(path);
    },
  };
}

export function readConfig(host: Tree) {
  const sourceText = host.read(CONFIG_PATH)?.toString('utf-8') || '';
  return JSON.parse(sourceText);
}

export function writeConfig(host: Tree, config: JSON) {
  host.overwrite(CONFIG_PATH, JSON.stringify(config, null, 2));
}

export function getAngularAppConfig(
  config: any,
  projectName: string
): any | never {
  if (!config.projects.hasOwnProperty(projectName)) {
    throw new SchematicsException(`Could not find project: ${projectName}`);
  }

  return config.projects[projectName];
}

export async function getProjectName(tree: Tree) {
  const host = createHost(tree);
  const { workspace } = await workspaces.readWorkspace('/', host);
  return workspace.extensions.defaultProject as string;
}
