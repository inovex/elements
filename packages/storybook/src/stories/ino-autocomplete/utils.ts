import { KeyValue } from '@inovex.de/elements';

export const AutoCompleteStoryOptions: KeyValue[] = [
  {
    key: 'hh',
    value: 'Hamburg',
  },
  {
    key: 'ber',
    value: 'Berlin',
  },
  {
    key: 'k',
    value: 'Karlsruhe',
  },
];
