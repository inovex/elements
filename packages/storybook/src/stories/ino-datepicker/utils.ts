import '@inovex.de/elements';

export function registerInlineDatepickerHandler() {
  const group = document.querySelector('ino-radio-group') as HTMLInoRadioGroupElement;
  const datepicker = document.querySelector('.datepicker-group ino-datepicker') as HTMLInoDatepickerElement;

  const today = document.querySelector('.today');
  const thisWeek = document.querySelector('.thisWeek');
  const lastWeek = document.querySelector('.lastWeek');

  const todayClickHandler = () => {
    datepicker.value = '02.03.2020';
    datepicker.label = 'Am';
    group.value = 'at';
    datepicker.range = false;
  };

  const thisWeekClickHandler = () => {
    group.value = 'range';
    datepicker.label = 'Zeitraum';
    datepicker.range = true;
    datepicker.value = '02.03.2020 bis 08.03.2020';
  };

  const lastWeekClickHandler = () => {
    group.value = 'range';
    datepicker.label = 'Zeitraum';
    datepicker.range = true;
    datepicker.value = '24.02.2020 bis 01.03.2020';
  };

  const checkedChangeHandler = (e: CustomEvent<boolean>) => {
    const target = e.target as HTMLInoRadioElement;

    switch (target.value) {
      case 'at':
        datepicker.label = 'Am';
        datepicker.range = false;
        break;
      case 'after':
        datepicker.label = 'Ab';
        datepicker.range = false;
        if (group.value == 'range') {
          datepicker.value = datepicker.value.split(' - ').shift();
        }
        break;
      case 'before':
        datepicker.label = 'Bis';
        datepicker.range = false;
        if (group.value == 'range') {
          datepicker.value = datepicker.value.split(' - ').pop();
        }
        break;
      case 'range':
        datepicker.range = true;
        datepicker.label = 'Zeitraum';
        break;
    }
    group.value = target.value;
  };

  today.addEventListener('click', todayClickHandler);
  thisWeek.addEventListener('click', thisWeekClickHandler);
  lastWeek.addEventListener('click', lastWeekClickHandler);
  group.addEventListener('checkedChange', checkedChangeHandler);

  return () => {
    today.removeEventListener('click', todayClickHandler);
    thisWeek.removeEventListener('click', thisWeekClickHandler);
    lastWeek.removeEventListener('click', lastWeekClickHandler);
    group.removeEventListener('checkedChange', checkedChangeHandler);
  };
}
