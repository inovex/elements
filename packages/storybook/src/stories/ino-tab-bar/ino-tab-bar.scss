.story-tab-bar {
  .customizable-tab {
    display: block;
    margin-left: auto;
    margin-right: auto;
    margin-bottom: 40px;
    align-self: center;
  }

  .story-tab-variants {
    display: flex;
    justify-content: space-around;
    flex-wrap: wrap;
  }
}
