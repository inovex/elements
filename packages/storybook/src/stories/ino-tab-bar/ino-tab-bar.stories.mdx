import { Meta, Description } from "@storybook/addon-docs";
import readme from "../../../../elements/src/components/ino-tab-bar/readme.md";

<Meta title="Structure/ino-tab-bar/Documentation" />

<Description markdown={readme} />
