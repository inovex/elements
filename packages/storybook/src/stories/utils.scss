/**
 * Applies styling to the container of a story.
 * The styling is encapsulated within this mixin because the ID may change in the future.
 */
@mixin story-container {
  #root-inner {
    @content;
  }
}
