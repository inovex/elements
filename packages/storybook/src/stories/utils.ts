import { html } from 'lit-html';
import ICONS from '../../../elements/src/components/ino-icon/icons';

export const maybeCreateStoryArgs = (story: any) => {
  if (!story.args) {
    story.args = {};
  }
  if (!story.argTypes) {
    story.argTypes = {};
  }
};

export const withIconControl = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    control: {
      type: 'select',
    },
    options: getIcons(),
  };
};

export const withColorScheme = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    control: {
      type: 'select',
    },
    options: [
      'primary',
      'secondary',
      'success',
      'warning',
      'error',
      'light',
      'dark',
    ],
  };
};

export const withSortDirection = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    options: ['desc', 'asc'],
    control: {
      type: 'select',
    },
  };
};

export const getIcons = () => ICONS.filter((icon) => !icon.startsWith('_'));

export const defaultDecorator = (story, className?: string) => {
  return html` <div class="${className ?? ''}">${story()}</div> `;
};
