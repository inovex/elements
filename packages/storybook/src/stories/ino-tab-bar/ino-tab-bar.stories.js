import CoreEvents from '@storybook/core-events';

import { withActions } from '@storybook/addon-actions';
import { text, boolean, number, select } from '@storybook/addon-knobs';

import withStencilReadme from '_local-storybookcore/with-stencil-readme';

import tabBarReadme from '_local-elements/src/components/ino-tab-bar/readme.md';
import tabReadme from '_local-elements/src/components/ino-tab/readme.md';
import ICONS from '_local-elements/src/components/ino-icon/icons';

import './ino-tab-bar.scss';
import addons from '@storybook/addons';

function subscribeToComponentEvents() {
  // == event block
  const eventHandler = function(e) {
    const el = e.target;

    if (!el) return;

    el.setAttribute('ino-active-tab', e.detail);
  };

  document.addEventListener('activeTabChange', eventHandler);
  // == event block

  // unsubscribe function will be called by Storybook
  return () => {
    document.removeEventListener('activeTabChange', eventHandler);
  };
}

export default {
  title: 'Structure/<ino-tab-bar>',

  decorators: [
    withStencilReadme(tabBarReadme),
    withActions('activeTabChange ino-tab-bar'),
    story => {
        addons
          .getChannel()
          .emit(CoreEvents.REGISTER_SUBSCRIPTION, subscribeToComponentEvents);
        return story();
      },
  ],
};

export const DefaultUsage = () => /*html*/ `
<div class="story-tab">
<h4>Customizable ino-tab-bar</h4>
  <div>
    <ino-tab-bar id="tab-bar-1" ino-active-tab="${number(
  'ino-active-tab',
  0
)}">
      <ino-tab ino-label="Label1" ino-icon="user" ino-stacked="${boolean(
  'ino-stacked',
  false
)}" ino-indicator-content-width="${boolean(
  'ino-indicator-content-width',
  false
)}"></ino-tab>
      <ino-tab ino-label="Label2" ino-icon="info" ino-stacked="${boolean(
  'ino-stacked',
  false
)}" ino-indicator-content-width="${boolean(
  'ino-indicator-content-width',
  false
)}"></ino-tab>
      <ino-tab ino-label="Label3" ino-icon="star" ino-stacked="${boolean(
  'ino-stacked',
  false
)}" ino-indicator-content-width="${boolean(
  'ino-indicator-content-width',
  false
)}"></ino-tab>
      <ino-tab ino-label="Label4" ino-icon="help" ino-stacked="${boolean(
  'ino-stacked',
  false
)}" ino-indicator-content-width="${boolean(
  'ino-indicator-content-width',
  false
)}"></ino-tab>
    </ino-tab-bar>
  </div>
  <h4>Stacked</h4>
  <ino-tab-bar>
    <ino-tab
            ino-label="Label 1"
            ino-icon="info"
            ino-stacked="true">
    </ino-tab>
    <ino-tab
            ino-label="Label 2"
            ino-icon="info"
            ino-stacked="true">
    </ino-tab>
    <ino-tab
            ino-label="Label 3"
            ino-icon="info"
            ino-stacked="true">
    </ino-tab>
    <ino-tab
            ino-label="Label 4"
            ino-icon="info"
            ino-stacked="true">
    </ino-tab>
  </ino-tab-bar>
  <h4>Indicator Content Width</h4>
  <ino-tab-bar>
    <ino-tab
            ino-label="Label 1"
            ino-icon="info"
            ino-indicator-content-width="true">
    </ino-tab>
    <ino-tab
            ino-label="Label 2"
            ino-icon="info"
            ino-indicator-content-width="true">
    </ino-tab>
    <ino-tab
            ino-label="Label 3"
            ino-icon="info"
            ino-indicator-content-width="true">
    </ino-tab>
    <ino-tab
            ino-label="Label 4"
            ino-icon="info"
            ino-indicator-content-width="true">
    </ino-tab>
  </ino-tab-bar>
</div>
`;

export const Tab = () => /*html*/ `
<div class="story-tab">
<h4>Customizable Tab</h4>
<div class="customizable-tab">
  <ino-tab
    ino-label="${text('ino-label', 'Customizable Tab')}"
    ino-icon="${select('ino-icon', ICONS, 'info')}"
    ino-stacked="${boolean('ino-stacked', false)}">
  </ino-tab>
  </div>
  <h4>Variants</h4>
  <div class="story-tab-variants">
    <ino-tab
        ino-label="Default"
        ino-icon="info"
        ino-stacked="false">
    </ino-tab>
    <ino-tab
          ino-label="Stacked"
          ino-icon="info"
          ino-stacked="true">
    </ino-tab>
    <ino-tab-bar ino-active-tab="0">
      <ino-tab
            ino-label="Default active Tab"
            ino-icon="info"
            >
      </ino-tab>
    </ino-tab-bar>
    <ino-tab-bar ino-active-tab="0">
      <ino-tab
            ino-label="Active content width indicator"
            ino-icon="info"
            ino-indicator-content-width="true"
            >
      </ino-tab>
    </ino-tab-bar>
  </div>
</div>
`;

Tab.decorators = [
  withStencilReadme(tabReadme)
];
