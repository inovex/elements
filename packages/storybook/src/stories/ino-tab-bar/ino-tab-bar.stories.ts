import { html } from 'lit-html';
import { defaultDecorator } from '../utils';
import { useEffect } from '@storybook/client-api';
import './ino-tab-bar.scss';
import { Story } from '@storybook/web-components';
import { Components } from '@inovex.de/elements';

const eventHandler = (e) => e.target.setAttribute('active-tab', e.detail);

export default {
  title: `Structure/ino-tab-bar`,
  component: 'ino-tab-bar',
  decorators: [
    defaultDecorator,
    (story) => {
      useEffect(() => {
        const tabBars = document.querySelectorAll('ino-tab-bar');
        tabBars.forEach((t) =>
          t.addEventListener('activeTabChange', eventHandler)
        );
        return () =>
          tabBars.forEach((t) =>
            t.removeEventListener('activeTabChange', eventHandler)
          );
      });
      return story();
    },
  ],
};

export const Playground: Story<Components.InoTabBar> = (args) => html`
  <ino-tab-bar id="customizable-tabbar" active-tab="${args.activeTab}">
    <ino-tab label="User" icon="user"></ino-tab>
    <ino-tab label="Messages" icon="message"></ino-tab>
    <ino-tab label="Settings" icon="settings"></ino-tab>
    <ino-tab label="Download" icon="download"></ino-tab>
  </ino-tab-bar>
`;
Playground.args = {
  activeTab: 0,
};
Playground.argTypes = {
  activeTab: {
    control: {
      type: 'select',
    },
    options: [0, 1, 2, 3],
  },
};

export const Stacked = () => html`
  <ino-tab-bar>
    <ino-tab label="Label 1" icon="info" stacked></ino-tab>
    <ino-tab label="Label 2" icon="info" stacked></ino-tab>
    <ino-tab label="Label 3" icon="info" stacked></ino-tab>
    <ino-tab label="Label 4" icon="info" stacked></ino-tab>
  </ino-tab-bar>
`;

export const IndicatorContentWidth = () => html`
  <ino-tab-bar>
    <ino-tab label="Label 1" icon="info" indicator-content-width></ino-tab>
    <ino-tab label="Label 2" icon="info" indicator-content-width></ino-tab>
    <ino-tab label="Label 3" icon="info" indicator-content-width></ino-tab>
    <ino-tab label="Label 4" icon="info" indicator-content-width></ino-tab>
  </ino-tab-bar>
`;
