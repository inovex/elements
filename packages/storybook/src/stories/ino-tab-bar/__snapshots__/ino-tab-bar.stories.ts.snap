// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Structure/ino-tab-bar ActiveTab smoke-test 1`] = `
<div id="root-inner">
  <div class="story-tab-bar">
    <ino-tab-bar id="customizable-tabbar"
                 active-tab="1"
                 auto-focus="false"
    >
      <ino-tab label="User"
               icon="user"
      >
      </ino-tab>
      <ino-tab label="Messages"
               icon="message"
      >
      </ino-tab>
      <ino-tab label="Settings"
               icon="settings"
      >
      </ino-tab>
      <ino-tab label="Download"
               icon="download"
      >
      </ino-tab>
    </ino-tab-bar>
  </div>
</div>
`;

exports[`Structure/ino-tab-bar Playground smoke-test 1`] = `
<div id="root-inner">
  <div class="story-tab-bar">
    <ino-tab-bar id="customizable-tabbar"
                 active-tab="0"
                 auto-focus="false"
    >
      <ino-tab label="User"
               icon="user"
      >
      </ino-tab>
      <ino-tab label="Messages"
               icon="message"
      >
      </ino-tab>
      <ino-tab label="Settings"
               icon="settings"
      >
      </ino-tab>
      <ino-tab label="Download"
               icon="download"
      >
      </ino-tab>
    </ino-tab-bar>
  </div>
</div>
`;
